src/chess/gui/Table.java

```diff
@@ -141,9 +151,34 @@ private Table() {
 
     private void setBoardDirection(final BoardDirection boardDirection) { this.boardDirection = boardDirection; }
 
+    /**
+     * Void Function that implements the two Timers
+     */
+    private void setupTimer(){
+        w = new Timer(this, whitePlayerSeconds, "white");
+        b = new Timer(this, blackPlayerSeconds, "black");
+
+        whitePlayer = new Thread(w);
+        blackPlayer = new Thread(b);
+
+        whitePlayerSeconds = time;
+        blackPlayerSeconds = time;
+
+        whitePlayer.start();
+        blackPlayer.start();
+        blackPlayer.suspend();
+
+        this.getGameSetup().whitePlayer = whitePlayer;
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [SA: Self assignment of value (SA_SELF_ASSIGNMENT)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#sa-self-assignment-of-field-sa-field-self-assignment).
This method contains a self assignment of a field; e.g.
Such assignments are useless, and may indicate a logic error or typo.

Reply:
Ya I did noticed that too but forgot to point out as well, thank you aimieball for pointing that out